src/Symfony/Component/HttpFoundation/EmptyResponse.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation;
+
+class EmptyResponse extends Response
+{
+    /**
+     * @param array $headers
+     */
+    public function __construct(array $headers = [])
+    {
+        parent::__construct(null, Response::HTTP_NO_CONTENT, $headers);
+    }
+
+    /**
+     * Factory method for chainability.
+     *
+     * Example:
+     *
+     *     return Response::create()
+     *         ->setSharedMaxAge(300);
+     *
+     * @param array $headers An array of response headers
+     *
+     * @return static
+     */
+    public static function create($content = '', $status = 200, $headers = [])
+    {
+        return new static($headers);
```

Comment:
https://github.com/symfony/symfony/issues/29326#issuecomment-443496518

so here we are :D We shouldnt ignore these params IMHO and throw a BadMethodCall perhaps, while always favoring `new EmptyResponse()`.

But really.. i mean `$empty =  new Response(204, '')` :man_shrugging:

Reply:
Oops, I did not saw this conversation (and your answer :smile: ) but yes, actually I was not sure on how to implement this method. Maybe some other people can help ?

Reply:
for this proposal to be viable i dont think we should abuse the Response inheritance tree. An EmptyResponse is just a Response with a specific status code and empty body.

As such i see 2 approaches:
- `Response::createEmpty($headers = []): self` (we dont care when one calls JsonResponse::createEmpty())
- `ResponseFactory::createEmpty(), createJson() etc.`

im not sure we want to introduce a new factory concept, so `Response::createEmpty()` looks good. Unless we decide to stick with `new Response('', 204)` instead. That's up to core members.